ivadomed/object_detection/utils.py

```diff
@@ -0,0 +1,268 @@
+import json
+import os
+import statistics
+
+import nibabel as nib
+import numpy as np
+from scipy import ndimage
+
+from ivadomed import postprocessing as imed_postpro
+from ivadomed import transforms as imed_transforms
+from ivadomed import utils as imed_utils
+from ivadomed.loader import utils as imed_loader_utils
+
+
+def get_bounding_boxes(mask):
+    """
+    Generates a 3D bounding box around a given mask.
+    :param mask: numpy array with the mask of the ROI
+    :return: bounding box coordinate (x_min, x_max, y_min, y_max, z_min, z_max)
+    """
+    # Label the different objects in the mask
+    labeled_mask, _ = ndimage.measurements.label(mask)
+    object_labels = np.unique(labeled_mask)
+    bounding_boxes = []
+    for label in object_labels[1:]:
+        single_object = labeled_mask == label
+        coords = np.where(single_object)
+        dimensions = []
+        for i in range(len(coords)):
+            dimensions.append(int(coords[i].min()))
+            dimensions.append(int(coords[i].max()))
+        bounding_boxes.append(dimensions)
+
+    return bounding_boxes
+
+
+def adjust_bb_size(bounding_box, factor, resample=False):
+    """
+    Modifies the bounding box dimensions according to a given factor
+    :param bounding_box: coordinates of bounding box (h_min, h_max, w_min, w_max, d_min, d_max)
```

Comment:
i find it confusing to alternate between h/w/d and x/y/z. I suggest to stick to one convention (x/y/z seems the most common approach in our current codebase)